### DISCUSSION TITLE
Publishing with a custom GitHub Actions workflow

### DISCUSSION THREAD

**Discussion by terual** [squidfunk/mkdocs-material#6686] [Q&A] [answered]
Has anybody already published a mkdocs-material site using a [custom GitHub Actions workflow](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow)? I am looking for a yaml GH Actions script which uses the GitHub Actions as a source for publishing instead of the standard "deploy from a branch".

There are some pointers [here](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site) but I can't seem to make it work.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @terual,
from the page with the pointers I see that the upload-pages-artifacts and deploy-pages actions are used.
The same is used in the workflow for this project's documentation:

https://github.com/squidfunk/mkdocs-material/blob/b6b531308e1c8516eca752a95353d827e1827066/.github/workflows/documentation.yml#L92-L112

So you should be able to use it as reference 🤔
